components/camel-jms/src/main/java/org/apache/camel/component/jms/JmsSendDynamicAware.java

```diff
@@ -109,8 +109,15 @@ private String parseDestinationName(String uri) {
         if (pos != -1) {
             uri = uri.substring(0, pos);
         }
-        // destination name is after last colon
-        pos = uri.lastIndexOf(':');
+
+        // destination name is after last colon (but not after double colon)
+        int posDoubleCol = uri.indexOf("::");
+        if (posDoubleCol != -1) {
+            pos = uri.substring(0, posDoubleCol).lastIndexOf(':');
+        } else {
+            pos = uri.lastIndexOf(':');
+        }
+
```

Comment:
Please, can you take a look at `StringHelper` (I guess `between` method) and try to use that, instead?

Reply:
Thanks, made use of `StringHelper`